### PR TITLE
typos: add to CI and fix all typos

### PR DESCRIPTION
**Pull request by phip1611**
add `typos` to CI and fix all typos

### CODE REVIEW

File: .github/workflows/build-site.yml [hunk 1]
```diff
@@ -55,6 +55,10 @@ jobs:
         name: "Install misspell"
       - run: bin/misspell -error content
         name: "Check for common typos"
+      - run: bin/misspell -error content
+        name: "Check for common typos"
```
Comment (phil-opp) [inline .github/workflows/build-site.yml:59]: This looks like a merge artifact.

```suggestion
```

File: content/this-month/2022-03/index.md [hunk 2]
```diff
@@ -227,7 +227,7 @@ emulation, and starting programs from ELF files.
 
 We merged a [new Korean translation](https://github.com/JOE1994) of first post of [_Writing an OS in Rust_](https://os.phil-opp.com/) blog this month. Thanks a lot to [@JOE1994](https://github.com/JOE1994) for creating this translation and [@QuqqU](https://github.com/QuqqU) for reviewing it!
 
-We also received lots of smaller fixes, by [@MaxDesiatov](https://github.com/phil-opp/blog_os/pull/1080), [@alaincao](https://github.com/phil-opp/blog_os/pull/1089), [@Programatic](https://github.com/phil-opp/blog_os/pull/1096), [@ruhuang2001](https://github.com/phil-opp/blog_os/pull/1091), [@Hofer-Julian](https://github.com/phil-opp/blog_os/pull/1093), [@SilensAngelusNex](https://github.com/phil-opp/blog_os/pull/1094), and [@julien-me](https://github.com/phil-opp/blog_os/pull/1095). Thank you all for your contributions!
```
Comment (phil-opp) [inline content/this-month/2022-03/index.md:230]: That's not a typo, that's their GitHub username :D.

**Review by phil-opp — APPROVED** [review]
Thanks a lot!

File: content/this-month/2022-03/index.md (hunk 2 above)
Comment (phip1611) [reply content/this-month/2022-03/index.md]: fixed

**Comment by phil-opp** [comment]
Thanks!

**Pull request merged**